SwiftCompilerSources/Sources/SIL/SmallProjectionPath.swift

```diff
@@ -262,7 +289,32 @@ public struct SmallProjectionPath : CustomStringConvertible, CustomReflectable,
       p = p.pop(numBits: numBits)
     }
   }
-  
+
+  /// Pops the last class projection and all following value fields from the tail of the path.
+  /// For example:
+  ///    `s0.e2.3.c4.s1` -> `s0.e2.3`
+  ///    `v**.c1.c4.s1`  -> `v**.c1`
+  ///    `c1.**`         -> `c1.**` (because it's unknown how many class projections are in `**`)
+  public func popLastClassAndValuesFromTail() -> SmallProjectionPath {
+    var p = self
+    var totalBits = 0
+    var neededBits = 0
+    while true {
+      let (k, _, numBits) = p.top
+      if k == .root { break }
+      if k.isClassField {
+        neededBits = totalBits
+        totalBits += numBits
+      } else {
+        totalBits += numBits
+        if !k.isValueField { neededBits = totalBits }
```

Comment:
`isValueField` is not `!isClassField` because `.anything` is not neither a class field nor a value field.
I'll add a comment.

Reply:
> isValueField is not !isClassField because .anything is not neither a class field nor a value field.

ok, sure, but isn't `isClassField` a subset of `!isValueField`?